File: 2-Ad-Hoc/1105-Sub-prime/1105.cpp
#include <iostream>

int main() {
    int nbanks, nfare, d, c, v;
    while (1) {
        std::cin>>nbanks>>nfare;
        if (nbanks == 0 && nfare == 0) break;
        int banks[nbanks];
        for (int i = 0; i < nbanks; ++i) std::cin>>banks[i];
        for (int i = 0; i < nfare; ++i) {
            std::cin>>d>>c>>v;
            banks[d-1] -= v;
            banks[c-1] += v;
        }
        bool able = true;
        for (int i = 0; i < nbanks; ++i) {
            if (banks[i] < 0) {
                able = false;
                break;
            }
        }
        if (able) {
            std::cout<<"S\n";
        }
        else {
            std::cout<<"N\n";
        }
    }
    return 0;
}

File: 2-Ad-Hoc/1105-Sub-prime/1105.py
while (1):
    nbanks, nfare = map(int, input().split())
    if nbanks == nfare == 0:
        break
    credit = input().split()
    banks = [int(i) for i in credit]
    for i in range (nfare):
        d, c, v = map(int, input().split())
        banks[d-1] -= v
        banks[c-1] += v
    able = True
    for i in banks:
        if i < 0:
            able = False
            break
    print ("S") if able else print ("N")